setup.py

```diff
@@ -496,6 +498,54 @@ def _read_requirements(filename: str) -> List[str]:
     return requirements
 
 
+def repackage_wheel(package_data: Dict[str, List[str]],
+                    wheel_location) -> None:
+    """Extracts libraries and other files from an existing wheel."""
+    assert _is_cuda(), "VLLM_USE_PRECOMPILED is only supported for CUDA builds"
+
+    import zipfile
+
+    if os.path.exists(wheel_filename := os.path.basename(wheel_location)):
```

Comment:
This assumes the wheel in current dir, and it will re-download the wheel when `VLLM_PRECOMPILED_WHEEL_LOCATION` is an abs path. Add a check like `os.path.isabs()`?

Reply:
Good catch, unsure why I used `wheel_filename`.

Used `os.path.isfile(wheel_location)` instead, which solves the problem as well.